lib/VMwareWebService/MiqVimVm.rb

```diff
@@ -751,7 +751,7 @@ def addDisk(backingFile, sizeInMB, label=nil, summary=nil)
 							# bck.diskMode = VirtualDiskMode::Independent_nonpersistent
 							bck.diskMode		= VirtualDiskMode::Independent_persistent
 						    bck.split			= "false"
-						    bck.thinProvisioned	= "false"
+						    bck.thinProvisioned	= thinProvisioned.to_s
```

Comment:
Then, if they pass in "false" it'll set it to "true". It's probably better to let it fail on the back-end.